web-extensions/chrome/scripts/load-bookmarks.js

```diff
@@ -141,11 +191,144 @@ function addEvents(mangaList) {
+            alert("BookMyComics does not support empty labels to identify a comic.<br>"
+                  + "Please define a label in the Side Panel's text area first.");
+        }
+
+        // Now do the actual registration
+        const handleRegisterError = err => {
+            bmcEngine.removeEventListener(bmcEngine.events.register.error, handleRegisterError);
+            bmcEngine.removeEventListener(bmcEngine.events.register.complete, handleRegisterSuccess);
+            mangaList.setMode(mangaList.MODE_BROWSE);
+            showHideSidePanel();
+            notifyResult('Register Comic', new Error('Could not alias current page to comicId ' + comicId));
+        };
+        const handleRegisterSuccess = () => {
+            bmcEngine.removeEventListener(bmcEngine.events.register.error, handleRegisterError);
+            bmcEngine.removeEventListener(bmcEngine.events.register.complete, handleRegisterSuccess);
+            mangaList.setMode(mangaList.MODE_BROWSE);
+            showHideSidePanel();
+            notifyResult('Register Comic');
+        };
+        bmcEngine.addEventListener(bmcEngine.events.register.error, handleRegisterError);
+        bmcEngine.addEventListener(bmcEngine.events.register.complete, handleRegisterSuccess);
+        // Don't need to provide the reader/name/chapter/pages, as they're
+        // already set in the engine since the instanciation.
+        bmcEngine.register(label);
+    };
+
+    bmcEngine._messaging.addWindowHandler(
+        bmcEngine.SIDEPANEL_ID,
+        evData => evData.type === 'action' && evData.action === 'notification',
+        evData => {
+            console.log(`BmcSidePanel: received message to display status notification op=${evData.operation} err=${evData.error}`);
+            notifyResult(evData.operation, evData.error);
+        });
 }
 
 var mangaList = new BmcMangaList();
 mangaList.generate();
 addEvents(mangaList);
 
+
+/*
+ * This function ensures the transition CSS class is absent, changes the
+ * background color (static, instantaneous).
+ * Then, it applies the transform class, and finally changes the color back to
+ * the original color. This creates a smooth transform from the provided color
+ * parameter to the original color.
+ */
+function triggerTransition(elem, color) {
+    // Ensure we get the final computed one (from CSS sheet), as using `.style`
+    // only accesses inline values.
+    const origColor = getComputedStyle(elem).backgroundColor;
+
+    // -> Ensure it's not present when setting the color
+    elem.classList.remove('notif-transform');
+
+    // Now set the color to transition from
+    elem.style.backgroundColor = color;
+
+    /*
+     * /!\ NOTE IMPORTANT /!\
+     * Accessing a DOM property seems to force a redraw, preventing browser
+     * optimization on style settings (by batching updates) which might
+     * actually hide the transition.
+     * /!\ NOTE IMPORTANT /!\
+     */
+    void elem.offsetHeight;
```

Comment:
Sounds quite complicated for doing a transition... Going through CSS and just changing the value of a CSS property wouldn't be easier? I'm pretty sure it doesn't impact rendering performances either.

Reply:
Well; we have an origin and target color.
As far as I understand, to transition from `A` to `B`,  one needs To have color `A` set; have the CSS transition class set; and only then apply the change of color to `B`.
First; I don't know how To do that in CSS Only. Maybe theres a back-and-forth transition fonction I should use to simplify..
Second, as I understand, the browser allows itself to batch the CSS property changes, which compacts all the color changes from this function into a Noop. The trick here prevents that mechanism by forcing the browser To redraw in between.

Any working simplification suggestion is thus welcome :)

Reply:
Huuuuum. I'll take a look later on but that sounds like something doable with CSS only (maybe I'm wrong though?).

Reply:
Let's address it as issue #18